Lib/typing.py

```diff
@@ -1237,6 +1242,12 @@ def __subclasscheck__(self, cls):
             if issubclass(cls, arg):
                 return True
 
+    def __reduce__(self):
+        func, (origin, args) = super().__reduce__()
+        if self._name == "Optional":
+            args = args[1] if args[0] is type(None) else args[0]
+        return func, (origin, args)
```

Comment:
You can simplify this down a little further:
```suggestion
        func, (origin, args) = super().__reduce__()
        return func, (Union, args)
```